### PR TITLE
Update API and modules to d.js v12 standards

### PR DESCRIPTION
**Pull request by davipatury**

### CODE REVIEW

**Comment by pedrofracassi** [comment]
DO NOT MERGE

**Comment by pedrofracassi** [comment]
This won't work if the guilds aren't in shard 0. Servers won't appear on the dashboard.

**Comment by Doges** [comment]
> 
> 
> This won't work if the guilds aren't in shard 0. Servers won't appear on the dashboard.

you tried it?

**Comment by pedrofracassi** [comment]
> > This won't work if the guilds aren't in shard 0. Servers won't appear on the dashboard.
> 
> you tried it?

A shard doesn't cache other shards' information because it doesn't receive that information from the gateway. If we're pulling the list of guilds in common from `guilds.cache`, we'll only get guilds managed by that shard.

File: src/http/api/guilds.js [hunk 1]
```diff
@@ -76,7 +77,7 @@ module.exports = class Guilds extends Route {
         }
       })
 
-    router.post('/:guildId/modules/:modName/methods/:methodName',
+    router.post('/:guildId(\\d{16,18})/modules/:modName/methods/:methodName',
```
Comment (pedrofracassi) [inline src/http/api/guilds.js:80]: oh wow, that's a thing?

**Comment by pedrofracassi** [comment]
I have implemented a pure-REST version of `/contributors` [here](https://github.com/SwitchbladeBot/api/blob/master/src/routes/contributors.js). Some pagination logic still needs to be added so it keeps working when we pass the 1000 guild members mark, though.

You can port it if you want :)

**Comment by pedrofracassi** [comment]
pls finish

**Comment by Doges** [comment]
demoramo tanto que saiu a v13 kkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkk

**Comment by Doges** [comment]
tanav13jakkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkk @next

**Pull request closed without merging**